tests/go-validation/v1apitests/provisioning_test.go

```diff
@@ -0,0 +1,92 @@
+package v1_api_tests
+
+import (
+	"fmt"
+	"testing"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/tests/go-validation/rancherrest"
+	"github.com/rancher/rancher/tests/go-validation/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	digitalOceanCloudCredentialName = "docloudcredential"
+	namespace                       = "fleet-default"
+	defaultRandStringLength         = 5
+	baseDOClusterName               = "docluster"
+)
+
+func TestProvisioning_DigitalOcean(t *testing.T) {
+	t.Log("Create Cloud Credential")
+	client, err := rancherrest.NewClient()
+	require.NoError(t, err)
+
+	cloudCredentialName := digitalOceanCloudCredentialName + utils.RandStringBytes(defaultRandStringLength)
+	doCloudCred := rancherrest.NewDigitalOceanCloudCredential(cloudCredentialName, "", namespace)
+
+	_, err = client.CreateCloudCredential(doCloudCred)
+	require.NoError(t, err)
+	t.Log("Cloud Credential was created successfully")
+
+	//randominize name
+	clusterName := baseDOClusterName + utils.RandStringBytes(defaultRandStringLength)
+
+	generatedPoolName := fmt.Sprintf("nc-%s-pool1", clusterName)
+
+	machinePoolConfig := rancherrest.NewMachinePoolConfig(generatedPoolName, rancherrest.DOKind, namespace, rancherrest.DOPoolType, "ubuntu-20-04-x64", "sfo3", "s-2vcpu-4gb")
```

Comment:
If we're using `us-east-2` as default Rancher server region I think it would be best to default to DO's `ny3` to reduce latency.

Reply:
Absolutely I'll make the change

Reply:
I also think something we should think about is creating enums for region, and others as well, so we don't have raw strings.